### PR TITLE
Allows agents to connect via a link-local IPv6 address

### PR DESCRIPTION
**Pull request by Nicogp**
|Related issue|Manual Testing|Integration Tests|Documentation|
|---|---|---|---|
|#15481|https://github.com/wazuh/wazuh-qa/issues/3667||https://github.com/wazuh/wazuh-documentation/pull/5826|

## Description

This PR adds the necessary changes to allow agents to connect via a link-local ipv6 address

### Added a new configuration option:
`<interface_index>`: sets the index by which the agent should try to connect to the server if the configured IP is a `link-local `IPv6. If the configured IP is not IPv6 link-local, this option has no effect.

This option is available on agent configuration, in the server block, and enrollment block:

```
<client>
    <server>
      <address>fe80::a00:27ff:feff:6b0b</address>
      <interface_index>3</interface_index>  <---- New option
      <port>1514</port>
      <protocol>tcp</protocol>
    </server>
    <enrollment>
      <enabled>yes</enabled>
      <manager_address>fe80::a00:27ff:feff:6b0b</manager_address>
      <interface_index>3</interface_index>  <---- New option
      <port>1515</port>
    </enrollment>
    <config-profile>ubuntu, ubuntu22, ubuntu22.04</config-profile>
    <notify_time>10</notify_time>
    <time-reconnect>60</time-reconnect>
    <auto_restart>yes</auto_restart>
    <crypto_method>aes</crypto_method>
  </client>
```

If one or more than one <server> blocks are configured, all with IPv6 link-local addresses, and none have interface_index configured, an error is logged and the agent fails to start:

Error example:
```
<client>
    <server>
      <address>fe80::a00:27ff:feff:6b0b</address>
      <port>1514</port>
      <protocol>tcp</protocol>
    </server>
    <server>
      <address>fe80::a00:27ff:feff:6b0b</address>
      <port>1514</port>
      <protocol>tcp</protocol>
    </server>
    <config-profile>ubuntu, ubuntu22, ubuntu22.04</config-profile>
    <notify_time>10</notify_time>
    <time-reconnect>60</time-reconnect>
    <auto_restart>yes</auto_restart>
    <crypto_method>aes</crypto_method>
  </client>
```

```
/var/ossec/bin/wazuh-control start
2022/12/15 22:46:15 wazuh-agentd: ERROR: (4114): All server addresses are IPv6 link-local and no interface to any <server> block has been configured.
2022/12/15 22:46:15 wazuh-agentd: CRITICAL: (1215): No client configured. Exiting.
wazuh-agentd: Configuration error. Exiting
```

### New parameter for the agent-auth tool
Also, a new parameter was added to the agent-auth tool, this is the "-n" parameter:
-n <network-interface>: Network interface to use in an IPv6 connection (only necessary in case of use of link-local address).

```
# /var/ossec/bin/agent-auth -h
 
Wazuh v4.4.0 - Wazuh Inc. (info@wazuh.com)
http://www.wazuh.com
  agent-auth: -[Vhdti] [-g group] [-D dir] [-m IP address] [-p port] [-n network-interface] [-A name] [-c ciphers] [-v path] [-x path] [-k path] [-P pass] [-G group] [-I IP address]
    -V                     Version and license message.
    -h                     This help message.
    -d                     Execute in debug mode. This parameter
                           can be specified multiple times
                           to increase the debug level.
    -t                     Test configuration.
    -g <group>             Group to run as (default: wazuh).
    -D <dir>               Directory to chdir into (default: /var/ossec).
    -m <addr>              Manager IP address.
    -p <port>              Manager port (default: 1515).
    -n <network-interface> Network interface to use in an IPv6 connection (only necessary in case of use of link-local address).
    -A <name>              Agent name (default: hostname).
    -c <cipher>            SSL cipher list (default: HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH)
    -v <path>              Full path to CA certificate used to verify the server.
    -x <path>              Full path to agent certificate.
    -k <path>              Full path to agent key.
    -P <pass>              Authorization password.
    -a                     Auto select SSL/TLS method. Default: TLS v1.2 only.
    -G <group>             Assigns the agent to one or more existing groups (separated by commas).
    -I <IP>                Set the agent IP address.
    -i                     Let the agent IP address be set by the manager connection.
```

```
# /var/ossec/bin/agent-auth -m fe80::a00:27ff:feff:6b0b -n 3 -A Test-agent-auth
2022/12/16 09:01:30 agent-auth: INFO: Started (pid: 10241).
2022/12/16 09:01:30 agent-auth: INFO: Requesting a key from server: FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B
2022/12/16 09:01:30 agent-auth: INFO: No authentication password provided
2022/12/16 09:01:30 agent-auth: INFO: Using agent name as: Test-agent-auth
2022/12/16 09:01:30 agent-auth: INFO: Waiting for server reply
2022/12/16 09:01:30 agent-auth: INFO: Valid key received
```

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language
- [x] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Coverity [#report](https://github.com/wazuh/wazuh/pull/15685#issuecomment-1355542218)
  - [x] Valgrind (memcheck and descriptor leaks check) [#report](https://github.com/wazuh/wazuh/pull/15685#issuecomment-1354044679)
- Memory tests for Windows
  - [x] Scan-build report
- Memory tests for macOS
  - [x] Scan-build report

- [x] Retrocompatibility with older Wazuh versions
- [x] Working on cluster environments
- [x] The data flow works as expected (agent-manager-api-app)
- [x] Added unit tests (for new features)

### CODE REVIEW

**Comment by Nicogp** [comment]
## Valgrind
For this test, two blocks of servers were configured, the first with an unreachable IP and the second with the correct IP of the administrator, in addition, after the agent connected, it was removed from the administrator forcing it to a new registration.

### Configuration
```
<client>
    <server>
      <address>fe80::a00:27ff:feff:1111</address>
      <port>1514</port>
      <protocol>tcp</protocol>
    </server>
    <server>
      <address>fe80::a00:27ff:feff:6b0b</address>
      <interface_index>3</interface_index>
      <port>1514</port>
      <protocol>tcp</protocol>
    </server>
    <config-profile>ubuntu, ubuntu22, ubuntu22.04</config-profile>
    <notify_time>10</notify_time>
    <time-reconnect>60</time-reconnect>
    <auto_restart>yes</auto_restart>
    <crypto_method>aes</crypto_method>
  </client>
```
### Valgrind report
```
# valgrind --leak-check=full --track-fds=yes /var/ossec/bin/wazuh-agentd -f
==1531190== Memcheck, a memory error detector
==1531190== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1531190== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==1531190== Command: /var/ossec/bin/wazuh-agentd -f
==1531190== 
2022/12/15 22:03:06 wazuh-agentd: INFO: (1410): Reading authentication keys file.
2022/12/15 22:03:06 wazuh-agentd: INFO: Using notify time: 10 and max time to reconnect: 60
2022/12/15 22:03:06 wazuh-agentd: INFO: Version detected -> Linux |nico-VirtualBox |5.15.0-56-generic |#62-Ubuntu SMP Tue Nov 22 19:54:14 UTC 2022 |x86_64 [Ubuntu|ubuntu: 22.04 (Jammy Jellyfish)] - Wazuh v4.4.0
2022/12/15 22:03:06 wazuh-agentd: INFO: Started (pid: 1531190).
2022/12/15 22:03:06 wazuh-agentd: INFO: Using AES as encryption method.
2022/12/15 22:03:06 wazuh-agentd: INFO: Trying to connect to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:1111]:1514/tcp).
2022/12/15 22:03:06 wazuh-agentd: INFO: No network interface provided to use with link-local IPv6 address.
2022/12/15 22:03:06 wazuh-agentd: ERROR: (1216): Unable to connect to '[FE80:0000:0000:0000:0A00:27FF:FEFF:1111]:1514/tcp': 'Invalid argument'.
2022/12/15 22:03:16 wazuh-agentd: INFO: Trying to connect to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:1111]:1514/tcp).
2022/12/15 22:03:16 wazuh-agentd: INFO: No network interface provided to use with link-local IPv6 address.
2022/12/15 22:03:16 wazuh-agentd: ERROR: (1216): Unable to connect to '[FE80:0000:0000:0000:0A00:27FF:FEFF:1111]:1514/tcp': 'Invalid argument'.
2022/12/15 22:03:26 wazuh-agentd: INFO: Trying to connect to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:1111]:1514/tcp).
2022/12/15 22:03:26 wazuh-agentd: INFO: No network interface provided to use with link-local IPv6 address.
2022/12/15 22:03:26 wazuh-agentd: ERROR: (1216): Unable to connect to '[FE80:0000:0000:0000:0A00:27FF:FEFF:1111]:1514/tcp': 'Invalid argument'.
2022/12/15 22:03:36 wazuh-agentd: INFO: Trying to connect to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:1111]:1514/tcp).
2022/12/15 22:03:36 wazuh-agentd: INFO: No network interface provided to use with link-local IPv6 address.
2022/12/15 22:03:36 wazuh-agentd: ERROR: (1216): Unable to connect to '[FE80:0000:0000:0000:0A00:27FF:FEFF:1111]:1514/tcp': 'Invalid argument'.
2022/12/15 22:03:46 wazuh-agentd: INFO: Trying to connect to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:1111]:1514/tcp).
2022/12/15 22:03:46 wazuh-agentd: INFO: No network interface provided to use with link-local IPv6 address.
2022/12/15 22:03:46 wazuh-agentd: ERROR: (1216): Unable to connect to '[FE80:0000:0000:0000:0A00:27FF:FEFF:1111]:1514/tcp': 'Invalid argument'.
2022/12/15 22:03:46 wazuh-agentd: INFO: Requesting a key from server: FE80:0000:0000:0000:0A00:27FF:FEFF:1111
2022/12/15 22:03:46 wazuh-agentd: INFO: No network interface provided to use with link-local IPv6 address.
2022/12/15 22:03:46 wazuh-agentd: ERROR: (1208): Unable to connect to enrollment service at '[FE80:0000:0000:0000:0A00:27FF:FEFF:1111]:1515'
2022/12/15 22:03:56 wazuh-agentd: WARNING: (4101): Waiting for server reply (not started). Tried: 'FE80:0000:0000:0000:0A00:27FF:FEFF:1111'.
2022/12/15 22:03:56 wazuh-agentd: INFO: Trying next server ip in the line: 'FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B'.
2022/12/15 22:03:56 wazuh-agentd: INFO: Trying to connect to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:03:57 wazuh-agentd: INFO: (4102): Connected to the server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:05:29 wazuh-agentd: ERROR: (1137): Lost connection with manager. Setting lock.
2022/12/15 22:05:29 wazuh-agentd: INFO: Closing connection to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:05:29 wazuh-agentd: INFO: Trying to connect to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:05:29 wazuh-agentd: WARNING: Process locked due to agent is offline. Waiting for connection...
2022/12/15 22:05:39 wazuh-agentd: INFO: Closing connection to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:05:39 wazuh-agentd: INFO: Trying to connect to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:05:49 wazuh-agentd: INFO: Closing connection to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:05:49 wazuh-agentd: INFO: Trying to connect to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:05:59 wazuh-agentd: INFO: Closing connection to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:05:59 wazuh-agentd: INFO: Trying to connect to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:06:09 wazuh-agentd: INFO: Closing connection to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:06:09 wazuh-agentd: INFO: Trying to connect to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:06:09 wazuh-agentd: INFO: Requesting a key from server: FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B
2022/12/15 22:06:10 wazuh-agentd: INFO: No authentication password provided
2022/12/15 22:06:10 wazuh-agentd: INFO: Using agent name as: nico-VirtualBox
2022/12/15 22:06:10 wazuh-agentd: INFO: Waiting for server reply
2022/12/15 22:06:10 wazuh-agentd: INFO: Valid key received
2022/12/15 22:06:10 wazuh-agentd: INFO: Waiting 20 seconds before server connection
2022/12/15 22:06:30 wazuh-agentd: INFO: (1410): Reading authentication keys file.
2022/12/15 22:06:30 wazuh-agentd: INFO: Closing connection to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:06:30 wazuh-agentd: INFO: Trying to connect to server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:06:30 wazuh-agentd: INFO: (4102): Connected to the server ([FE80:0000:0000:0000:0A00:27FF:FEFF:6B0B]:1514/tcp).
2022/12/15 22:06:30 wazuh-agentd: INFO: Server responded. Releasing lock.
2022/12/15 22:06:34 wazuh-agentd: INFO: Agent is now online. Process unlocked, continuing...
2022/12/15 22:10:54 wazuh-agentd: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
==1531190== 
==1531190== FILE DESCRIPTORS: 8 open (3 std) at exit.
==1531190== Open AF_INET6 socket 7: [fe80::e08d:22a8:9550:23d6]:33398 <-> [fe80::a00:27ff:feff:6b0b]:1514
==1531190==    at 0x529CCEB: socket (syscall-template.S:120)
==1531190==    by 0x1BBBA4: OS_Connect (os_net.c:259)
==1531190==    by 0x1BBE0E: OS_ConnectTCP (os_net.c:328)
==1531190==    by 0x11CAA4: connect_server (start_agent.c:94)
==1531190==    by 0x11D477: agent_handshake_to_server (start_agent.c:333)
==1531190==    by 0x11CD00: start_agent (start_agent.c:158)
==1531190==    by 0x114CA0: AgentdStart (agentd.c:193)
==1531190==    by 0x117E44: main (main.c:190)
==1531190== 
==1531190== Open file descriptor 6: queue/rids/sender_counter
==1531190==    at 0x5289764: open (open64.c:41)
==1531190==    by 0x5201135: _IO_file_open (fileops.c:188)
==1531190==    by 0x5201491: _IO_file_fopen@@GLIBC_2.2.5 (fileops.c:280)
==1531190==    by 0x51F472D: __fopen_internal (iofopen.c:75)
==1531190==    by 0x51F472D: fopen@@GLIBC_2.2.5 (iofopen.c:86)
==1531190==    by 0x187FCE: OS_StartCounter (msgs.c:94)
==1531190==    by 0x1860B4: OS_UpdateKeys (keys.c:446)
==1531190==    by 0x11D33A: try_enroll_to_server (start_agent.c:307)
==1531190==    by 0x11CCE6: start_agent (start_agent.c:157)
==1531190==    by 0x114CA0: AgentdStart (agentd.c:193)
==1531190==    by 0x117E44: main (main.c:190)
==1531190== 
==1531190== Open AF_UNIX socket 5: <unknown>
==1531190==    at 0x529CCEB: socket (syscall-template.S:120)
==1531190==    by 0x1BB9B5: OS_ConnectUnixDomain (os_net.c:211)
==1531190==    by 0x1A1705: StartMQWithSpecificOwnerAndPerms (mq_op.c:39)
==1531190==    by 0x1A17F4: StartMQ (mq_op.c:61)
==1531190==    by 0x114957: AgentdStart (agentd.c:130)
==1531190==    by 0x117E44: main (main.c:190)
==1531190== 
==1531190== Open AF_UNIX socket 4: queue/sockets/queue
==1531190==    at 0x529CCEB: socket (syscall-template.S:120)
==1531190==    by 0x1BB71B: OS_BindUnixDomainWithPerms (os_net.c:149)
==1531190==    by 0x1A164F: StartMQWithSpecificOwnerAndPerms (mq_op.c:30)
==1531190==    by 0x1A17F4: StartMQ (mq_op.c:61)
==1531190==    by 0x1146B9: AgentdStart (agentd.c:75)
==1531190==    by 0x117E44: main (main.c:190)
==1531190== 
==1531190== Open file descriptor 3: /dev/urandom
==1531190==    at 0x52896EB: open (open64.c:41)
==1531190==    by 0x1A4EAA: randombytes (randombytes.c:62)
==1531190==    by 0x1A4F92: srandom_init (randombytes.c:82)
==1531190==    by 0x1143B7: AgentdStart (agentd.c:27)
==1531190==    by 0x117E44: main (main.c:190)
==1531190== 
==1531190== 
==1531190== HEAP SUMMARY:
==1531190==     in use at exit: 52,557 bytes in 53 blocks
==1531190==   total heap usage: 43,275 allocs, 43,222 frees, 969,670,530 bytes allocated
==1531190== 
==1531190== 288 bytes in 1 blocks are possibly lost in loss record 44 of 52
==1531190==    at 0x484DA83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==1531190==    by 0x40147D9: calloc (rtld-malloc.h:44)
==1531190==    by 0x40147D9: allocate_dtv (dl-tls.c:375)
==1531190==    by 0x40147D9: _dl_allocate_tls (dl-tls.c:634)
==1531190==    by 0x520A834: allocate_stack (allocatestack.c:430)
==1531190==    by 0x520A834: pthread_create@@GLIBC_2.34 (pthread_create.c:647)
==1531190==    by 0x1A3614: CreateThreadJoinable (pthreads_op.c:47)
==1531190==    by 0x1A36C5: CreateThread (pthreads_op.c:62)
==1531190==    by 0x1147FD: AgentdStart (agentd.c:106)
==1531190==    by 0x117E44: main (main.c:190)
==1531190== 
==1531190== 288 bytes in 1 blocks are possibly lost in loss record 45 of 52
==1531190==    at 0x484DA83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==1531190==    by 0x40147D9: calloc (rtld-malloc.h:44)
==1531190==    by 0x40147D9: allocate_dtv (dl-tls.c:375)
==1531190==    by 0x40147D9: _dl_allocate_tls (dl-tls.c:634)
==1531190==    by 0x520A834: allocate_stack (allocatestack.c:430)
==1531190==    by 0x520A834: pthread_create@@GLIBC_2.34 (pthread_create.c:647)
==1531190==    by 0x1A3614: CreateThreadJoinable (pthreads_op.c:47)
==1531190==    by 0x1A36C5: CreateThread (pthreads_op.c:62)
==1531190==    by 0x11485A: AgentdStart (agentd.c:114)
==1531190==    by 0x117E44: main (main.c:190)
==1531190== 
==1531190== 288 bytes in 1 blocks are possibly lost in loss record 46 of 52
==1531190==    at 0x484DA83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==1531190==    by 0x40147D9: calloc (rtld-malloc.h:44)
==1531190==    by 0x40147D9: allocate_dtv (dl-tls.c:375)
==1531190==    by 0x40147D9: _dl_allocate_tls (dl-tls.c:634)
==1531190==    by 0x520A834: allocate_stack (allocatestack.c:430)
==1531190==    by 0x520A834: pthread_create@@GLIBC_2.34 (pthread_create.c:647)
==1531190==    by 0x1A3614: CreateThreadJoinable (pthreads_op.c:47)
==1531190==    by 0x1A36C5: CreateThread (pthreads_op.c:62)
==1531190==    by 0x1148D6: AgentdStart (agentd.c:121)
==1531190==    by 0x117E44: main (main.c:190)
==1531190== 
==1531190== 288 bytes in 1 blocks are possibly lost in loss record 47 of 52
==1531190==    at 0x484DA83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==1531190==    by 0x40147D9: calloc (rtld-malloc.h:44)
==1531190==    by 0x40147D9: allocate_dtv (dl-tls.c:375)
==1531190==    by 0x40147D9: _dl_allocate_tls (dl-tls.c:634)
==1531190==    by 0x520A834: allocate_stack (allocatestack.c:430)
==1531190==    by 0x520A834: pthread_create@@GLIBC_2.34 (pthread_create.c:647)
==1531190==    by 0x1A3614: CreateThreadJoinable (pthreads_op.c:47)
==1531190==    by 0x1A36C5: CreateThread (pthreads_op.c:62)
==1531190==    by 0x114A15: AgentdStart (agentd.c:150)
==1531190==    by 0x117E44: main (main.c:190)
==1531190== 
==1531190== LEAK SUMMARY:
==1531190==    definitely lost: 0 bytes in 0 blocks
==1531190==    indirectly lost: 0 bytes in 0 blocks
==1531190==      possibly lost: 1,152 bytes in 4 blocks
==1531190==    still reachable: 51,405 bytes in 49 blocks
==1531190==         suppressed: 0 bytes in 0 blocks
==1531190== Reachable blocks (those to which a pointer was found) are not shown.
==1531190== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==1531190== 
==1531190== For lists of detected and suppressed errors, rerun with: -s
==1531190== ERROR SUMMARY: 4 errors from 4 contexts (suppressed: 0 from 0)
==1531190== could not unlink /tmp/vgdb-pipe-from-vgdb-to-1531190-by-root-on-???
==1531190== could not unlink /tmp/vgdb-pipe-to-vgdb-from-1531190-by-root-on-???
==1531190== could not unlink /tmp/vgdb-pipe-shared-mem-vgdb-1531190-by-root-on-???
```

File: src/config/client-config.c [hunk 1]
```diff
@@ -586,3 +616,47 @@ bool Validate_Address(agent_server *servers)
 
     return false;
 }
+
+/* Checks if at least one <server> block is not a link-local ipv6 address or it has a network interface configured. */
+bool Validate_IPv6_Link_Local_Interface(agent_server *servers) {
+    int i;
```
Comment (cborla) [inline src/config/client-config.c:622]: It could be an `unsigned int`

**Review by cborla — DISMISSED** [review]
LGTM!

**Comment by Nicogp** [comment]
## Coverity

Coverity reported no new issues:

image
image

File: src/config/client-config.c [hunk 2]
```diff
@@ -586,3 +616,47 @@ bool Validate_Address(agent_server *servers)
...
+        /* The hostname was not resolved correctly */
+        if (ip_address == NULL || *ip_address == '\0') {
+            if (servers[i].rip != NULL) {
+                const int rip_l = strlen(servers[i].rip);
+                mdebug1("Could not resolve hostname '%.*s'", servers[i].rip[rip_l - 1] == '/' ? rip_l - 1 : rip_l, servers[i].rip);
+            } else {
+                mdebug1("Could not resolve hostname");
+            }
+            os_free(ip_address);
+            return true;
```
Comment (TomasTurina) [inline src/config/client-config.c:645]: Maybe continue?

**Review by cborla — APPROVED** [review]
LGTM!

**Review by TomasTurina — APPROVED** [review]

**Review by jmv74211 — APPROVED** [review]
## QA review

- **Type**: Manual testing.
- **Status**: Approved  🟢 
- **Testing issue**: https://github.com/wazuh/wazuh-qa/issues/3667
- **Comments**:  https://github.com/wazuh/wazuh-qa/issues/3667#issuecomment-1363085075

**Pull request merged**